plugin/evm/ftso_system_test.go

```diff
@@ -126,6 +127,283 @@ func TestFTSOSystem_Contracts(t *testing.T) {
 	})
 }
 
+func TestFTSOSystem_Details(t *testing.T) {
+	initTestContracts(t)
+
+	auth, be := simulatedBlockchain(t)
+	defer be.Close()
+
+	vpInt := big.NewInt(100)
+
+	unclaimedReward := big.NewInt(200)
+
+	rewardEpochDurationSeconds := big.NewInt(301)
+	rewardEpochPowerHeight := big.NewInt(302)
+	rewardEpochStartHeight := big.NewInt(303)
+	rewardEpochStartTime := big.NewInt(304)
+	rewardEpochsStartTs := big.NewInt(1) // this should be less than header.Time()
+
+	index1 := big.NewInt(123)
+	index2 := big.NewInt(321)
+	supportedIndices := []*big.Int{index1, index2}
+
+	indices := []*big.Int{big.NewInt(401)}
+	priceProvidersAddress1 := common.BigToAddress(big.NewInt(402))
+	priceProvidersAddresses := []common.Address{priceProvidersAddress1}
+
+	providerAddresses := []common.Address{common.BigToAddress(big.NewInt(500))}
+	nodes := [][20]byte{{1, 2}}
+
+	// VotePower contract
+	votePowerAddr := deployTestContract(t, auth, be, testAbiVotePower, testAbiVotePowerBin, vpInt)
+
+	// Wnat contract
+	wnatAddr := deployTestContract(t, auth, be, testAbiWnat, testAbiWnatBin, votePowerAddr)
+
+	// Reward contract
+	rewardAddr := deployTestContract(t, auth, be, testAbiReward, testAbiRewardBin, wnatAddr, unclaimedReward)
+
+	// Manager contract
+	ftsoManagerAddr := deployTestContract(t, auth, be, testAbiManager, testAbiManagerBin, rewardAddr, rewardEpochDurationSeconds, rewardEpochPowerHeight, rewardEpochStartHeight, rewardEpochStartTime, rewardEpochsStartTs)
+
+	// Registry contract
+	ftsoRegistryAddr := deployTestContract(t, auth, be, testAbiFtsoRegistry, testAbiFtsoRegistryBin, supportedIndices)
+
+	// Whitelist contract
+	voterWhitelisterAddr := deployTestContract(t, auth, be, testAbiVoterWhitelister, testAbiVoterWhitelisterBin, indices, priceProvidersAddresses)
+
+	// Validation contract
+	validationAddr := deployTestContract(t, auth, be, testAbiValidation, testAbiValidationBin, providerAddresses, nodes)
+
+	// Submitter contract
+	submitterAddr := deployTestContract(t, auth, be, testAbiSubmitter, testAbiSubmitterBin, voterWhitelisterAddr, ftsoRegistryAddr, ftsoManagerAddr)
+
+	be.Commit(true)
+
+	ftsoSystem := testFTSOSystem(t, be, submitterAddr, validationAddr)
+
+	epoch := big.NewInt(1).Uint64()
+	details, err := ftsoSystem.Details(epoch)
+	require.NoError(t, err)
+
+	assert.Equal(t, rewardEpochPowerHeight.Uint64(), details.PowerHeight)
+	assert.Equal(t, rewardEpochStartHeight.Uint64(), details.StartHeight)
+	assert.Equal(t, rewardEpochStartTime.Uint64(), details.StartTime)
+	assert.Equal(t, rewardEpochStartTime.Uint64()+rewardEpochDurationSeconds.Uint64(), details.EndTime)
+}
+
+func TestFTSOSystem_Snapshot(t *testing.T) {
+	initTestContracts(t)
+
+	t.Run("common case", func(t *testing.T) {
```

Comment:
```suggestion
	t.Run("nominal case", func(t *testing.T) {
```

A nitpick, but for consistency :)